sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py

```diff
@@ -46,6 +54,43 @@ def setUp(self):
     def tearDown(self):
         super(AppConfigurationClientTest, self).tearDown()
 
+    @app_config_decorator
+    def test_mock_policies(self, client):
+        from azure.core.pipeline.transport import HttpRequest, HttpResponse, HttpTransport
+        from azure.core.pipeline.policies import RetryPolicy
+        from azure.core.pipeline import Pipeline, PipelineResponse
+
+        class MockTransport(HttpTransport):
+            def __init__(self):
+                self._count = 0
+                self.auth_headers = []
+            def __exit__(self, exc_type, exc_val, exc_tb):
+                pass
+            def close(self):
+                pass
+            def open(self):
+                pass
+
+            def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+                self._count += 1
+                self.auth_headers.append(
+                    request.headers['Authorization']
+                )
+                response = HttpResponse(request, None)
+                response.status_code = 429
+                return response
+
+        http_request = HttpRequest('GET', 'http://aka.ms/')
+        transport = MockTransport()
+
+        pipeline = Pipeline(transport, client._policies)
```

Comment:
We can create a new client here rather than use the one created in the test bed. :)

Reply:
@xiangyan99 I'm curious why we want to create a new client? The client in the test bed uses the default policies.

Reply:
But it does not use our custom transport. :)